lms/djangoapps/email_marketing/tests/test_signals.py

```diff
@@ -459,9 +491,97 @@ def test_update_unenrolled_list(self, mock_sailthru_client):
+        """
+        mock_sailthru_client.api_get.return_value = SailthruResponse(JsonResponse({'lists': []}))
+        _get_or_create_user_list(mock_sailthru_client, 'test1_user_list')
+        mock_sailthru_client.api_get.assert_called_with("list", {})
+        mock_sailthru_client.api_post.assert_called_with(
+            "list", {'list': 'test1_user_list', 'primary': 0, 'public_name': 'test1_user_list'}
+        )
+
+        # test existing user list
+        mock_sailthru_client.api_get.return_value = \
+            SailthruResponse(JsonResponse({'lists': [{'name': 'test1_user_list'}]}))
+        _get_or_create_user_list(mock_sailthru_client, 'test2_user_list')
+        mock_sailthru_client.api_get.assert_called_with("list", {})
+        mock_sailthru_client.api_post.assert_called_with(
+            "list", {'list': 'test2_user_list', 'primary': 0, 'public_name': 'test2_user_list'}
+        )
+
+        # test get error from Sailthru
+        mock_sailthru_client.api_get.return_value = \
+            SailthruResponse(JsonResponse({'error': 43, 'errormsg': 'Got an error'}))
+        self.assertEqual(_get_or_create_user_list(
+            mock_sailthru_client, 'test1_user_list'), None
+        )
+
+        # test post error from Sailthru
+        mock_sailthru_client.api_post.return_value = \
+            SailthruResponse(JsonResponse({'error': 43, 'errormsg': 'Got an error'}))
+        mock_sailthru_client.api_get.return_value = SailthruResponse(JsonResponse({'lists': []}))
+        self.assertEqual(_get_or_create_user_list(mock_sailthru_client, 'test2_user_list'), None)
+
+    @patch('email_marketing.tasks.SailthruClient')
+    def test_get_sailthru_list_map_no_list(self, mock_sailthru_client):
+        """Test when no list returned from sailthru"""
+        mock_sailthru_client.api_get.return_value = SailthruResponse(JsonResponse({'lists': []}))
+        self.assertEqual(_get_list_from_email_marketing_provider(mock_sailthru_client), {})
+        mock_sailthru_client.api_get.assert_called_with("list", {})
+
+    @patch('email_marketing.tasks.SailthruClient')
+    def test_get_sailthru_list_map_error(self, mock_sailthru_client):
+        """Test when no list returned from sailthru"""
+        mock_sailthru_client.api_get.return_value = SailthruResponse(
+            JsonResponse({'error': 43, 'errormsg': 'Got an error'})
+        )
+        self.assertEqual(_get_list_from_email_marketing_provider(mock_sailthru_client), {})
+
+    @patch('email_marketing.tasks.SailthruClient')
+    def test_get_sailthru_list_map_exception(self, mock_sailthru_client):
+        """Test when no list returned from sailthru"""
+        mock_sailthru_client.api_get.side_effect = SailthruClientError
+        self.assertEqual(_get_list_from_email_marketing_provider(mock_sailthru_client), {})
+
+    @patch('email_marketing.tasks.SailthruClient')
+    def test_get_sailthru_list(self, mock_sailthru_client):
+        """Test when no list returned from sailthru"""
```

Comment:
The comments are the same for this test case, and the two others above it. Same comments like this adds no value. Please either add the docstring to make these tests distinct, or remove these docstrings if you think the code are clear enough